### PR TITLE
[improve][test] Add subscribing regex topic test for `delete_when_subscriptions_caught_up`.

### PR DESCRIPTION
**Pull request by Technoboy-**
### Motivation

Add subscribing regex topic test for https://github.com/apache/pulsar/pull/18283.

### Documentation

- [ ] `doc` 
- [ ] `doc-required` 
- [x] `doc-not-needed` 
- [ ] `doc-complete`

### CODE REVIEW

File: pulsar-broker/src/test/java/org/apache/pulsar/broker/service/InactiveTopicDeleteTest.java [hunk 1]
```diff
@@ -324,20 +324,33 @@ public void testDeleteWhenNoBacklogs() throws Exception {
             .subscriptionName("sub")
             .subscribe();
 
-        for (int i = 0; i < 10; i++) {
+        Consumer<byte[]> consumer2 = pulsarClient.newConsumer()
+                .topicsPattern("persistent://prop/ns-abc/test.*")
```
Comment (mattisonchao) [inline pulsar-broker/src/test/java/org/apache/pulsar/broker/service/InactiveTopicDeleteTest.java:328]: Question:
Shall we create multiple topics to test?

**Review by mattisonchao — APPROVED** [review]

File: pulsar-broker/src/test/java/org/apache/pulsar/broker/service/InactiveTopicDeleteTest.java (hunk 1 above)
Comment (Technoboy-) [reply pulsar-broker/src/test/java/org/apache/pulsar/broker/service/InactiveTopicDeleteTest.java:332]: added.

File: pulsar-broker/src/test/java/org/apache/pulsar/broker/service/InactiveTopicDeleteTest.java [hunk 2]
```diff
@@ -314,30 +315,50 @@ public void testDeleteWhenNoBacklogs() throws Exception {
...
 
         admin.topics().skipAllMessages(topic, "sub");
-        Awaitility.await()
-                .untilAsserted(() -> Assert.assertFalse(admin.topics().getList("prop/ns-abc").contains(topic)));
+        Awaitility.await().untilAsserted(() -> {
+            Assert.assertFalse(admin.topics().getList("prop/ns-abc").contains(topic));
+            Assert.assertFalse(admin.topics().getList("prop/ns-abc").contains(topic2));
+        });
         consumer.close();
+        consumer2.close();
```
Comment (codelipenghui) [inline pulsar-broker/src/test/java/org/apache/pulsar/broker/service/InactiveTopicDeleteTest.java:361]: We should check that consumer2 has no internal consumers eventually or we can say all the internal consumers are closed due to the topic being deleted.

Comment (Technoboy-) [reply pulsar-broker/src/test/java/org/apache/pulsar/broker/service/InactiveTopicDeleteTest.java:363]: updated

Comment (Technoboy-) [reply pulsar-broker/src/test/java/org/apache/pulsar/broker/service/InactiveTopicDeleteTest.java:363]: Ah, line 358 has checked all the consumers.

**Review by congbobo184 — APPROVED** [review]

**Review by gaoran10 — COMMENTED** [review]
Sorry, I have a little confused, why the topic can be deleted? It seems the topic has connected consumers.

**Comment by Technoboy-** [comment]
> Sorry, I have a little confused, why the topic can be deleted? It seems the topic has connected consumers.

Ah, some motivation and modification can be found here https://github.com/apache/pulsar/pull/18283
I also update the related doc.

**Review by poorbarcode — APPROVED** [review]

**Review by gaoran10 — APPROVED** [review]

**Review by codelipenghui — APPROVED** [review]

**Pull request merged**